apps/server/src/shared/common/validator/date.validator.ts

```diff
@@ -0,0 +1,12 @@
+import { ValidatorConstraint, ValidatorConstraintInterface, ValidationArguments } from 'class-validator';
+
+@ValidatorConstraint({ name: 'isBefore', async: false })
+export class IsBeforeConstraint implements ValidatorConstraintInterface {
+	validate(propertyValue: Date, args: ValidationArguments) {
+		return propertyValue < args.object[args.constraints[0] as unknown as string];
+	}
+
+	defaultMessage(args: ValidationArguments) {
+		return `${args.property} must be before ${args.constraints[0] as unknown as string}`;
```

Comment:
also regarding the logic itself: I know it does not make user sense, but why dont we provide the case the available date == due date? the current logic would prohibit such behaviour, but this would also mean that if we have both values  empty, it would not go through the validation, right?!

Reply:
the validator actually does not work with optional fields...so if the dueDate is empty, it fails
was a nice idea, but I guess I to move this validation in the use case / service instead....